pypi_resource/out.py

```diff
@@ -81,7 +80,9 @@ def out(srcdir, input):
         raise NamesValidationError(
             f"Different names for package ({package_name}) and input ({input_name}). If this is intentional, you can configure `name_must_match` to `false`."
         )
-    if not is_canonical(version):
+
+    semver = common.py_version_to_semver(str(version))
```

Comment:
This is actually a converter, if the version is valid, it returns its semver version, but we don't care about its value. We just want to check that is not None.

Maybe replacing this part with:
```
try:
  pipio.Version(str(version))
except pipio.InvalidVersion:
   raise VersionValidationError(
            f"Version {version} string is not compliant with PEP 440 versioning convention"
        )
```

would be better.

Reply:
Because that way `packaging` can make incompatible changes without `pip` having to care.

Another explanation would be that this way `pip` is as stand-alone as can be and so you cannot break it by (force) installing an incompatible `packaging` package. That allows you to still use `pip` to recover from a broken state.

Reply:
I added the commit with try/except directly. It seems clearer. 

Also, thanks for explanation.